src/metabase/models/params/field_values.clj

```diff
@@ -100,26 +101,24 @@
     :impersonation
     (field-values/hash-key-for-impersonation field-id)))
 
-(defn create-advanced-field-values!
-  "Fetch and create a FieldValues for `field` with type `fv-type`.
-  The human_readable_values of Advanced FieldValues will be automatically fixed up based on the
-  list of values and human_readable_values of the full FieldValues of the same field."
+(defn prepare-advanced-field-values
+  "Fetch and construct the FieldValues for `field` with type `fv-type`. This does not do any insertion.
+   The human_readable_values of Advanced FieldValues will be automatically fixed up based on the
+   list of values and human_readable_values of the full FieldValues of the same field."
   [fv-type field hash-key constraints]
-  (when-let [{wrapped-values :values
-              :keys [has_more_values]} (fetch-advanced-field-values fv-type field constraints)]
+  (when-let [{wrapped-values :values :keys [has_more_values]}
```

Comment:
Aside: I found it very confusing that we'll only create advanced field values if we can fetch them. It turns out that the language here is very overloaded and this means "calculate them from the underlying fields". Perhaps it's worth renaming, but this whole side quest has gotten me exhausted enough already.

Reply:
A further thought, scrolling past this again - I wonder whether we shouldn't cache the absence of values anyway. It might still be quite expensive to do the underlying query.